build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -0,0 +1,280 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fetch from "node-fetch";
+import { promises as fsPromises } from "fs";
+import { Flags } from "@oclif/core";
+import { Logger } from "@fluidframework/build-tools";
+import { BaseCommand } from "../../base";
+import { PackageVersionList } from "../../lib";
+import { isInternalVersionRange } from "@fluid-tools/version-tools";
+
+// Define the interface for build details
+interface IBuildDetails {
+	definition: { name: string };
+	status: string;
+	result: string;
+	sourceBranch: string;
+	finishTime: string;
+	buildNumber: string;
+}
+
+export class UnreleasedReportCommand extends BaseCommand<typeof UnreleasedReportCommand> {
+	static description = `Creates a release report for the most recent build published to an internal ADO feed. It does this by finding the most recent build in ADO produced from a provided branch, and creates a report using that version. The report always uses the "caret" report format.`;
```

Comment:
Couple of questions...

1. The command only works for the client release group, correct?
2. The version ranges in the report will have exact ranges for client packages, and caret versions for everything else (like server packages and independent packages), correct?

If those are both true, consider this:

```suggestion
	static description = `Creates a release report for the most recent build of the client release group published to an internal ADO feed. It does this by finding the most recent build in ADO produced from a provided branch, and creates a report using that version. The report is a combination of the "simple" and "caret" report formats. Packages released as part of the client release group will have an exact version range, while other packages, such as server packages or independent packages, will have a caret-equivalent version range.`;
```

Reply:
Yes. I'll update the description.